src/button-group.ts

```diff
@@ -194,77 +194,49 @@ export default class ButtonGroup extends LitElement {
   }
 
   #onDefaultSlotKeydown(event: KeyboardEvent) {
-    const selectedButtonElement = this.querySelector(
-      'glide-core-button-group-button[selected]',
+    const enabledButtons = this.#buttonElements.filter(
+      ({ disabled }) => !disabled,
     );
 
-    switch (event.key) {
-      case 'ArrowUp':
-      case 'ArrowLeft': {
-        // Prevent page scroll.
-        event.preventDefault();
-
-        let previousButton =
-          selectedButtonElement?.previousElementSibling ??
-          this.#buttonElements.at(-1);
-
-        while (previousButton instanceof ButtonGroupButton) {
-          if (previousButton.disabled) {
-            previousButton =
-              previousButton.previousElementSibling ??
-              this.#buttonElements.at(-1);
-          } else {
-            break;
-          }
-        }
+    const selectedButtonIndex = enabledButtons.findIndex(
+      ({ selected }) => selected,
+    );
 
-        if (previousButton instanceof ButtonGroupButton) {
-          previousButton.privateSelect();
-        }
+    if (['ArrowUp', 'ArrowLeft'].includes(event.key)) {
+      const previousButton =
+        enabledButtons[
+          (selectedButtonIndex - 1 + enabledButtons.length) %
```

Comment:
Though slightly obtuse, using the remainder operator is apparently the most common way to get the previous and next item in array with wrapping. I don't love it. But it's shorter and, to my mind at least, no less easier to understand than what I replaced. It's also simpler get test coverage for.